### PR TITLE
feat(connectors): add universal source and destination using PyAirbyte

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary

Adds universal source and destination connectors that leverage PyAirbyte as a backend. These connectors are named `destination-pyairbyte` and `source-pyairbyte` internally, but will be published to the Airbyte connector registry as `destination-smoke-test` and `source-smoke-test` (via a companion PR in airbytehq/airbyte).

**Universal Destination** (`destination-pyairbyte`): Writes to any PyAirbyte-supported cache backend (DuckDB, Postgres, Snowflake, BigQuery, MotherDuck). Implements the Airbyte CDK `Destination` interface with `spec()`, `check()`, `write()`, and support for `overwrite`/`append` sync modes.

**Universal Source** (`source-pyairbyte`): Wraps any PyAirbyte-supported source connector, acting as a proxy that delegates to the underlying source based on configuration. Implements the CDK `Source` interface with `spec()`, `check()`, `discover()`, and `read()` methods.

**Smoke Test Source** (`source-smoke-test`): A dedicated test source with predefined scenarios for destination regression testing. Includes scenarios for basic types, timestamps, large decimals, nested JSON, null handling, column naming edge cases, and more.

All connectors are registered as CLI entrypoints and include a Dockerfile for containerization.

## Updates since last revision

- Merged `source-smoke-test` connector from PR #982 (predefined scenarios for destination regression testing)
- Reverted CLI entrypoints back to `destination-pyairbyte`/`source-pyairbyte` (internal PyAirbyte naming)
- The airbyte connector registry will use `destination-smoke-test`/`source-smoke-test` names (handled in companion PR)
- Added TK-TODO comment for STATE message handling (tracks tech debt for incremental sync support)
- Imported TK-TODO CI check workflow from airbyte-ops-mcp repo

## Review & Testing Checklist for Human

- [ ] **State handling limitation (source)**: The `state` parameter in `read()` is ignored and STATE messages are skipped. Incremental syncs will not work - only full refresh is supported. This is tracked with a TK-TODO marker.
- [ ] **SQL injection risk (destination)**: Schema and table names are interpolated into SQL strings without identifier quoting. Verify this is acceptable given these values come from configuration/catalog.
- [ ] **Private API usage (source)**: Uses PyAirbyte's internal `_read_with_catalog()` method which may change without notice.
- [ ] **Exception messages in check()**: The source `check()` method returns raw exception text which could potentially expose config details. Consider if this needs sanitization.
- [ ] **No automated tests included**: Manual testing was performed but unit/integration tests should be added.

**Recommended test plan:**
1. Build the Docker image locally
2. **Destination**: Run a test sync with DuckDB backend, verify records in `_airbyte_raw_*` tables
3. **Source**: Configure with a simple source (e.g., `source-faker`), run discover and read operations
4. **Smoke Test Source**: Run `source-smoke-test` to verify predefined scenarios emit expected records
5. **Pipe test**: `source-pyairbyte read --config source.json --catalog catalog.json | destination-pyairbyte write --config dest.json --catalog catalog.json`

### Notes

This is part 1 of 2 PRs - a companion PR will register these connectors in the airbytehq/airbyte repo as `destination-smoke-test`.

Requested by: @aaronsteers (AJ)
Devin session: https://app.devin.ai/sessions/46cd19c4793a4e1ea32bb660de9c1fe0

### CODE REVIEW

File: airbyte/cli/universal_connector/source.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                    type=Type.RECORD,
+                    record=AirbyteRecordMessage(
+                        stream=message.record.stream,
+                        data=message.record.data,
+                        emitted_at=message.record.emitted_at,
+                        namespace=message.record.namespace,
+                    ),
+                )
+            # Skip STATE messages for now - they require per-stream format conversion
+            # which is complex and not needed for basic full-refresh syncs
```
Comment (aaronsteers) [inline airbyte/cli/universal_connector/source.py]: Add a TK-TODO comment here, and inport the TK-TODO CI check from the ops repo.

**Comment by aaronsteers** [comment]
Closing for now. Replaced by:

- #995 
- #859 
- #981
